transport/client.go

```diff
@@ -2,34 +2,141 @@ package transport
 	"google.golang.org/protobuf/proto"
+	"log/slog"
 	"math/rand"
+	"sync"
+	"time"
+)
+
+var (
+	ErrTimeout = errors.New("timeout connecting to radio")
 )
 
 type HandlerFunc func(message proto.Message)
 
 type Client struct {
 	sc       *StreamConn
 	handlers *HandlerRegistry
-	log      *log.Logger
-
-	config struct {
-		complete bool
-		configID uint32
-		*meshtastic.MyNodeInfo
-		*meshtastic.DeviceMetadata
-		nodes    []*meshtastic.NodeInfo
-		channels []*meshtastic.Channel
-		config   []*meshtastic.Config
-		modules  []*meshtastic.ModuleConfig
-	}
+	log      *slog.Logger
+
+	State State
+}
+
+type State struct {
+	sync.RWMutex
+	complete       bool
+	configID       uint32
+	nodeInfo       *meshtastic.MyNodeInfo
+	deviceMetadata *meshtastic.DeviceMetadata
+	nodes          []*meshtastic.NodeInfo
+	channels       []*meshtastic.Channel
+	configs        []*meshtastic.Config
+	modules        []*meshtastic.ModuleConfig
+}
+
+func (s *State) Complete() bool {
+	s.RLock()
+	defer s.RUnlock()
+	return s.complete
+}
+
+func (s *State) ConfigID() uint32 {
+	s.RLock()
+	defer s.RUnlock()
+	return s.configID
+}
+
+func (s *State) NodeInfo() *meshtastic.MyNodeInfo {
+	s.RLock()
+	defer s.RUnlock()
+	return s.nodeInfo
+}
+
+func (s *State) DeviceMetadata() *meshtastic.DeviceMetadata {
+	s.RLock()
+	defer s.RUnlock()
+	return s.deviceMetadata
+}
+
+func (s *State) Nodes() []*meshtastic.NodeInfo {
+	s.RLock()
+	defer s.RUnlock()
+	return s.nodes
+}
+
+func (s *State) Channels() []*meshtastic.Channel {
+	s.RLock()
+	defer s.RUnlock()
+	return s.channels
+}
+
+func (s *State) Configs() []*meshtastic.Config {
+	s.RLock()
+	defer s.RUnlock()
+	return s.configs
+}
+
+func (s *State) Modules() []*meshtastic.ModuleConfig {
+	s.RLock()
+	defer s.RUnlock()
+	return s.modules
+}
+
+func (s *State) SetComplete(complete bool) {
+	s.Lock()
+	defer s.Unlock()
+	s.complete = complete
+}
+
+func (s *State) SetConfigID(configID uint32) {
+	s.Lock()
+	defer s.Unlock()
+	s.configID = configID
+}
+
+func (s *State) SetNodeInfo(nodeInfo *meshtastic.MyNodeInfo) {
+	s.Lock()
+	defer s.Unlock()
+	s.nodeInfo = nodeInfo
+}
+
+func (s *State) SetDeviceMetadata(deviceMetadata *meshtastic.DeviceMetadata) {
+	s.Lock()
+	defer s.Unlock()
+	s.deviceMetadata = deviceMetadata
+}
+
+func (s *State) AddNode(node *meshtastic.NodeInfo) {
+	s.Lock()
+	defer s.Unlock()
+	s.nodes = append(s.nodes, node)
+}
+
+func (s *State) AddChannel(channel *meshtastic.Channel) {
+	s.Lock()
+	defer s.Unlock()
+	s.channels = append(s.channels, channel)
+}
+
+func (s *State) AddConfig(config *meshtastic.Config) {
+	s.Lock()
+	defer s.Unlock()
+	s.configs = append(s.configs, config)
+}
+
+func (s *State) AddModule(module *meshtastic.ModuleConfig) {
+	s.Lock()
+	defer s.Unlock()
+	s.modules = append(s.modules, module)
 }
 
 func NewClient(sc *StreamConn, errorOnNoHandler bool) *Client {
 	return &Client{
-		log:      log.WithPrefix("client"),
+		log:      slog.Default().WithGroup("client"),
```

Comment:
Ideally, we'll extend the NewClient method to allow a slog logger to be passed in and used. We can keep this as a default if none is provided though.

Reply:
Sounds good to me